File: Data Structures & Algorithms/max-water-container/submission-0.cs
public class Solution {
    public int MaxArea(int[] heights) {
        int max = 0;
 int capacity; 
 for(int i=0; i<heights.Length; i++)
 {
     for(int j =i+1;j<heights.Length; j++)
     {
         capacity=heights[j]< heights [i]? heights[j]*Math.Abs(i-j) :heights[i] * Math.Abs(i - j);
         max = Math.Max(max, capacity);
     }
 }
 return max; 
    }
}

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.cpp
class Solution {
public:
int largestRectangleArea(vector<int>& heights) {
	pair<int, int> top;
	vector<int> stack;
	int maxArea = 0;
	for (int i = 0; i < heights.size(); i++) {
		while (!stack.empty() && heights[stack.back()] > heights[i]) {
			top = { stack.back(), heights[stack.back()] };
			stack.pop_back();
			int area = top.second * (stack.empty() ? i : i - stack.back() - 1);
			maxArea = max(maxArea, area);
		}
		stack.push_back(i);
	}
	while (!stack.empty()) {
		top = { stack.back(), heights[stack.back()] };
		stack.pop_back();
		int area = top.second * (stack.empty() ? heights.size() : heights.size() - stack.back() - 1);
		maxArea = max(maxArea, area);
	}
	return maxArea;
}
};
